### PR TITLE
Category option combos support

### PR DESCRIPTION
**Pull request by mestachs**
Companion PR : https://github.com/BLSQ/orbf-rules_engine/pull/56
 
## Context

Add a category combo to "iterate/expand" over activity rules at the package level

image

Possible usage : spread an amount over multiple category combo 
as described in https://github.com/BLSQ/orbf-rules_engine/pull/56/files#diff-11e0072ab65f5cb0a8e5be83f73c264a

todo 
 - [ ] check non regression for other projects

known issues
  - simulation is ugly but displays values, perhaps better to show lines catcombo and catcombo names.
  - we are in a bit of a hurry

## Self proof reading checklist

- [x] Did I run Pronto and fixed all warnings (pronto run -c origin/dev)
- [x] Make sure all naming and code will remain understandable in 6 month by someone new to the project or by you.
- [x] Did I test the right thing?
- [?] Did I test corner cases, non happy path (defensive testing)?
- [x] Check that I used the ad-hoc patterns and created files accordingly (Presenters, Services, Search object, Form object,...)
- [x] Check code efficiency with record intensive project
- [x] Update documentation,readme accordingly

Thanks!

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -85,7 +85,7 @@ else
   # We're using both of them against the latest master. We should set
   # them to versions when they become more stable
   gem "hesabu", github: "BLSQ/hesabu"
-  gem "orbf-rules_engine", github: "BLSQ/orbf-rules_engine"
+  gem "orbf-rules_engine", github: "BLSQ/orbf-rules_engine", branch: "category_option_combos_support"
```
Comment (pjaspers) [inline Gemfile]: Which reminds me, now that Github is doing the gem hosting, we could start using them to do actual gem releases.

Also, for future me: this needs to move to regular orbf-rules_engine before we deploy this to master.

File: lib/data_test/capture.rb [hunk 2]
```diff
@@ -73,11 +73,12 @@ def dump_input_values(values)
       record_json(path_for_artefact("input-values", "json"), inputs)
     end
 
-    def dump_project(project, indicators)
+    def dump_project(project, _data_compound)
       project.dhis2_url = "https://redacted.example.com"
       project.user = "redacted"
       project.password = "redacted"
-      data = MapProjectToOrbfProject.new(project, indicators, engine_version).map
+      data = MapProjectToOrbfProject.new(project, indicators.indicators, indicators.category_combos,
```
Comment (pjaspers) [inline lib/data_test/capture.rb]: `indicators` is no longer defined here?

Comment (mestachs) [reply lib/data_test/capture.rb]: ... forgot to rename _data_compound => data_compound.indicators
